crud.go

```diff
@@ -387,12 +387,43 @@ func (bucket *WalrusBucket) GetXattr(k string, xattrKey string, xv interface{})
 	return 0, errors.New("GetXattr not implemented for walrus")
 }
 
-func (bucket *WalrusBucket) GetSubDocRaw(k string, subdocKey string)(value []byte, casOut uint64, err error){
-	return nil, 0, errors.New("GetSubDocRaw not implemented for walrus")
+// GetSubDocRaw Walrus implementation only works with a top-level subdocKey
+func (bucket *WalrusBucket) GetSubDocRaw(k string, subdocKey string) (value []byte, casOut uint64, err error) {
+	var fullDoc map[string]interface{}
+	casOut, err = bucket.Get(k, &fullDoc)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	if fullDoc[subdocKey] == nil {
+		return nil, 0, fmt.Errorf("SubdocGetRaw with key %s and subdocKey %s not found, %w", k, subdocKey, bucket.missingError(k))
+	}
+
+	value, err = json.Marshal(fullDoc[subdocKey])
+	return value, casOut, err
+
 }
 
-func (bucket *WalrusBucket) WriteSubDoc(k string, subdocKey string, cas uint64, value []byte)(casOut uint64, err error){
-	return 0, errors.New("WriteSubdoc not implemented for walrus")
+// WriteSubDoc Walrus implementation only works with a top-level subdocKey
+func (bucket *WalrusBucket) WriteSubDoc(k string, subdocKey string, cas uint64, value []byte) (casOut uint64, err error) {
+	// Get existing doc (if it exists) to change sub doc value in
+	fullDoc := make(map[string]interface{})
+	casOut, err = bucket.Get(k, &fullDoc)
+	if err != nil && !errors.Is(err, bucket.missingError(k)) {
+		return 0, err
+	}
+
+	// Set new subdoc value
+	var subDocVal map[string]interface{}
+	err = json.Unmarshal(value, &subDocVal)
+	fullDoc[subdocKey] = subDocVal
+
+	// Write full doc body to bucket
+	err = bucket.Set(k, 0, nil, fullDoc)
+	if err != nil {
+		return 0, err
+	}
+	return casOut, nil
```

Comment:
I think `casOut` should be the cas we got from the Write, otherwise people calling this can't modify the document in a CAS-safe way without an extra get.